### ISSUE TITLE
Do we need a resource map validator?

### ISSUE THREAD

**Issue by gothub** [DataONEorg/rdataone#148]
Do we need a function that validates a resource map before it is uploaded to a member node?
This hopefully isn't an issue for auto-generated resmaps. For resource maps that have been
hand edited, it would be nice to have some mechanism to validate them to ensure thae

- the XML is valid
- all objects referenced in the resmap exists on the target mn

Hand editing resmaps seems to have been ad-hoc, so maybe there aren't clear use cases for this

**Comment by amoeba** [comment]
I think this could be a great idea.

When would validation be triggered? Is this when a user calls createObject/updateObject and the sysmeta for that object indicates the formatID is `http://www.openarchives.org/ore/terms`? i.e. catching users that are accessing the lower-level API and making mistakes?

Re your second point above, I think there are two reasonable ways to verify whether an object exists on the MN:

1. Get the HTTP status code returned at /meta/{PID}
2. Search for the object by its PID using the Solr index

(1) doesn't scale well when the number of objects referenced in the resource map gets high and (2) gets tricky when (a) the objects could be inserted but not yet indexed or (b) when the objects are private. 

Just some first thoughts.

**Comment by gothub** [comment]
Yes, validation could be specified as a param to createObject/UpdateObject but also as a 
standalone function call. Could also be used to validate EML (when the EML R package is released).

The `describeObject` method could be used to check for the existence of a pid.
